dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -97,9 +98,9 @@ static tartype_t kTarArchiveType = {LibtarOpenTarArchive, [](int) -> int { retur
 
 TarArchive::TarArchive(std::unique_ptr<FileStream> stream_)
     : stream_(std::move(stream_)),
-      archiveoffset_(0),
       instance_handle_(Register(this)),
-      eof_(false) {
+      archiveoffset_(0),
+      eof_(this->stream_->Size() == 0) {
```

Comment:
This really shouldn't be in the initializer list unless it absolutely has to (and it doesn't, since it's a plain boolean) - calling member functions from the initializer list is asking for trouble.

Reply:
done